### ISSUE TITLE
feat(authoring): per-table optimizer warnings returned during authoring

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#761]
## Problem

`pkg/dtrules/decisiontable/analysis.go` defines `AnalyzeTable` with structural checks (no-op columns, subsumed columns, unreachable columns), but the only caller today is `analysis_test.go`. No authoring surface returns them, so the warnings never reach an author or agent editing tables via the JSON CLI (#716) or MCP server (#717).

Per-table warnings need to surface immediately while editing, scoped to the table being touched, so the authoring loop is responsive without running a full project pass. Project-wide analysis (every table, EDD checks, cross-table consistency, deployment gating) lives in the Full Review step (#768).

### Warning struct

Optimizer findings are always advisory — they never crash deployment. They are warnings, full stop. Errors (parse failures, EL compile errors, undefined references, structural problems) come from other validators and are reported separately by the Full Review.

Extend `decisiontable.Warning` (`analysis.go:23`) before any consumer ships:

```go
type Warning struct {
    Table        string `json:"table"`
    Column       int    `json:"column"`        // 1-based, 0 if not column-scoped
    ConditionRow int    `json:"condition_row"` // 0-based, -1 if not row-scoped
    Kind         string `json:"kind"`
    Reason       string `json:"reason"`
}
```

The struct grows now so #762 (entry-scoped) and #766 (row-scoped) don't churn it later. No `Severity` field: every entry is a warning by category, and the deployment gate (#768) treats them all the same way — informational, never blocking.

### Authoring channel

- Run `AnalyzeTable` on the single edited table during `table_get` / `table_put` / `table_patch` MCP responses. Embed the warnings array in the response payload.
- New MCP tool `table_warnings`: read-only fetch of warnings for a named table (parses + compiles the tree + runs checks + returns JSON). Cache keyed on the table's content hash so repeat calls are cheap.
- CLI: `dtrules table warnings <name>` returns the same JSON.

## Out of scope

- Project-wide / cross-table analysis — Full Review (#768).
- Error reporting (parse, EL compile, structural). Different category; flows through different validators into the Full Review.
- The individual new checks (#762/#763/#765/#766) — this issue establishes the channel; they plug in.

## Acceptance

- `table_get` on a table with a no-op column returns a `warnings: []` array in the response.
- `table_warnings` MCP tool returns warnings for a single named table.
- `decisiontable.Warning` has `ConditionRow` field with stable JSON tag.
- `make check` passes.

Part of the decision-table advisory-pass epic (#767).
